pkg/cmd/image/push.go

```diff
@@ -149,10 +150,16 @@ func Push(ctx context.Context, client *containerd.Client, rawRef string, options
 		return err
 	}
 
-	if err = signutil.Sign(rawRef, options.GOptions.Experimental, options.SignOptions); err != nil {
+	if img, err := client.ImageService().Get(ctx, ref); err != nil {
 		return err
+	} else {
+		if err = signutil.Sign(fmt.Sprintf("%s@%s", strings.SplitN(ref, ":", 2)[0],
```

Comment:
Doesn't seem to work when the ref contains a registry port number

Reply:
How to fix this 🤔

Reply:
https://pkg.go.dev/github.com/containerd/containerd/reference#Parse

Reply:
thank you, it seems that everything is fine.